Client/src/StoreModules/QuestionStoreModule.ts

```diff
@@ -617,22 +617,36 @@ async function loadQuestion(
   initialParamData?: Record<string, string>,
 ) {
   const step = stepId ? await wdkService.findStep(stepId) : undefined;
+  const initialParams = step ? initialParamDataFromStep(step) : initialParamData != null ? initialParamData : {};
+
   try {
-    const question = step == null
-      ? await wdkService.getQuestionAndParameters(searchName)
-      : await wdkService.getQuestionGivenParameters(searchName, step.searchConfig.parameters);
+    const question = Object.keys(initialParams).length > 0
+      ? await wdkService.getQuestionGivenParameters(searchName, initialParams)
```

Comment:
If I'm not mistaken, if `initialParams` contains values for only a subset of the search's parameters, validation for this call will fail. Previously, we would merge `initialParamData` with the search's default parameter values. It doesn't look like that is happening now. Can you confirm that this works? E.g., https://beta.plasmodb.org/plasmo.beta/app/search/pathway/PathwaysByGeneList?param.ds_gene_ids.idList=PF3D7_0321200 should load the search page with that one dataset id.

Reply:
I was able to test the above on my development site, and it does not work. The problem is that the param `ds_gene_ids` is a `datasetParam`, and the value of those params is a dataset record id (not to be confused with a _userDataset_).

The code is not parsing out the param name properly: it's using `ds_gene_ids.idList` instead of `ds_gene_ids`. There are two things to consider to address this issue.

First, this parameter is not a depended param, so we don't really need to make the call to the question service w/ the parameter value payload. We could add some additional logic to check if we actually need to make this call. This would be useful due to the next consideration.

Second, if this parameter were a depended parameter, we would need to first make a dataset record to get a dataset id to use in the payload of the question service request. This can potentially lead to generating a lot of dataset records.

Thus, I would recommend we first do a check to see if any of the params, for which an initial value is provided, is a depended param. To get this information, you need to get the `QuestionWithParameters` instance from the service (via `GET`). Each parameter has an array property `dependentParams`. If the array is empty, then it is not depended.

The following line could be used to check if we need to make a POST request to the question service:

    const initialParamsHasDependents = question.parameters.some(p => p.name in initialParams && p.dependentParams.length > 0);